snapshotter/volume_integ_test.go

```diff
@@ -0,0 +1,151 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/containerd/containerd"
+	"github.com/containerd/containerd/namespaces"
+	"github.com/containerd/containerd/oci"
+	"github.com/firecracker-microvm/firecracker-containerd/internal/integtest"
+	"github.com/firecracker-microvm/firecracker-containerd/proto"
+	"github.com/firecracker-microvm/firecracker-containerd/runtime/firecrackeroci"
+	"github.com/firecracker-microvm/firecracker-containerd/snapshotter/internal/integtest/stargz/fs/source"
+	"github.com/firecracker-microvm/firecracker-containerd/volume"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+const mib = 1024 * 1024
+
+func TestGuestVolumeFrom_Isolated(t *testing.T) {
+	integtest.Prepare(t, integtest.WithDefaultNetwork())
+
+	runtimes := []string{
+		"aws.firecracker",
+		// "io.containerd.runc.v2"
+	}
+
+	for _, rt := range runtimes {
+		t.Run(rt, func(t *testing.T) {
+			testGuestVolumeFrom(t, rt)
+		})
+	}
+}
+
+func testGuestVolumeFrom(t *testing.T, runtime string) {
+	const vmID = "default"
+	al2 := imageRef
+	postgres := "docker.io/library/postgres:14.3"
```

Comment:
`al2 := imageRef` doesn't seem necessary, as `imageRef` is a constant and `al2`'s value never changes.  I'd recommend either renaming the constant if clarity was the concern, or just using `imageRef` in place of `al2` directly.

For `postgres`, that can probably be a constant, like `vmID`.  There isn't much consistency with these assignments.

Reply:
LeLet me rename imageRef to al2stargz.